Source/Utility/Diagnostics/DiagProbe.cpp

```diff
@@ -0,0 +1,328 @@
+    amrex::Real* tmp_values_d = m_values_at_probe_d.data();
+    auto const& m_probe_idx_d = m_probe_idx;
+    auto const& low_cell_idx_d = low_cell_idx;
+
+    auto& m_probe_loc_d = m_probe_loc;
+    auto& x_low_cell_d = x_low_cell;
+    auto& dx_finest_lev_probe_d = dx_finest_lev_probe;
+    auto& m_interpType_d = m_interpType;
+
+    amrex::ParallelFor(
+      m_fieldIndices_d.size(), [=] AMREX_GPU_DEVICE(int n) noexcept {
+        amrex::Array3D<amrex::Real, 0, 1, 0, 1, 0, 1> neighbour_cells{
+          0.0}; // Neighbour cell solution values
+        int stIdx = idx_d_p[n];
+
+        if (m_interpType_d == Linear) {
+
+#if (AMREX_SPACEDIM == 1)
+          neighbour_cells(0, 0, 0) = state(low_cell_idx_d[0], 0, 0, stIdx);
+          neighbour_cells(1, 0, 0) = state(low_cell_idx_d[0] + 1, 0, 0, stIdx);
+#elif (AMREX_SPACEDIM == 2)
+          neighbour_cells(0, 0, 0) =
+            state(low_cell_idx_d[0], low_cell_idx_d[1], 0, stIdx);
+          neighbour_cells(1, 0, 0) =
+            state(low_cell_idx_d[0] + 1, low_cell_idx_d[1], 0, stIdx);
+          neighbour_cells(0, 1, 0) =
+            state(low_cell_idx_d[0], low_cell_idx_d[1] + 1, 0, stIdx);
+          neighbour_cells(1, 1, 0) =
+            state(low_cell_idx_d[0] + 1, low_cell_idx_d[1] + 1, 0, stIdx);
+#else
+          neighbour_cells(0, 0, 0) = state(
+            low_cell_idx_d[0], low_cell_idx_d[1], low_cell_idx_d[2], stIdx);
+          neighbour_cells(1, 0, 0) = state(
+            low_cell_idx_d[0] + 1, low_cell_idx_d[1], low_cell_idx_d[2], stIdx);
+          neighbour_cells(0, 1, 0) = state(
+            low_cell_idx_d[0], low_cell_idx_d[1] + 1, low_cell_idx_d[2], stIdx);
+          neighbour_cells(1, 1, 0) = state(
+            low_cell_idx_d[0] + 1, low_cell_idx_d[1] + 1, low_cell_idx_d[2],
+            stIdx);
+
+          neighbour_cells(0, 0, 1) = state(
+            low_cell_idx_d[0], low_cell_idx_d[1], low_cell_idx_d[2] + 1, stIdx);
+          neighbour_cells(1, 0, 1) = state(
+            low_cell_idx_d[0] + 1, low_cell_idx_d[1], low_cell_idx_d[2] + 1,
+            stIdx);
+          neighbour_cells(0, 1, 1) = state(
+            low_cell_idx_d[0], low_cell_idx_d[1] + 1, low_cell_idx_d[2] + 1,
+            stIdx);
+          neighbour_cells(1, 1, 1) = state(
+            low_cell_idx_d[0] + 1, low_cell_idx_d[1] + 1, low_cell_idx_d[2] + 1,
+            stIdx);
+#endif
+          tmp_values_d[n] = LinearInterpolate(
+            m_probe_loc_d, x_low_cell_d, neighbour_cells,
+            dx_finest_lev_probe_d);
+        } else if (m_interpType_d == CellCenter) {
+#if (AMREX_SPACEDIM == 1)
+          tmp_values_d[n] = state(m_probe_idx_d[0], 0, 0, stIdx);
+#elif (AMREX_SPACEDIM == 2)
+          tmp_values_d[n] = state(m_probe_idx_d[0], m_probe_idx_d[1], 0, stIdx);
+#else
+          tmp_values_d[n] =
+            state(m_probe_idx_d[0], m_probe_idx_d[1], m_probe_idx_d[2], stIdx);
+#endif
+        }
+      });
+  }
+
+  amrex::ParallelDescriptor::ReduceRealSum(
```

Comment:
this is the kinda thing that is crushing users in amr-wind (until I fixed it ;) ). Each rank is holding way more data than it actually needs. Just a comment, not something to fix here.